### DISCUSSION TITLE
stacksorm Extreme Workflow Composer Goes Down

### DISCUSSION THREAD

**Discussion by komalphadatare** [StackStorm/st2#5480] [Q&A] [answered]
Hi,
Stacksorm Extreme Workflow Composer Goes Down and shows Internal server error and it happens frequently.
while analyzing the logs i found that mongod is down due to which its not able to fetch the data to bring the service up.

**Comment by amanda11** [comment, marked as answer]
Take a look at why your mongod service failed. One problem I have seen on systems with not much memory, is that mongod can become killed by the kernel as over time it increases its heap size, and so if machine is low on memory will be the candidate process to get killed by the O/S. In that situation you will need to review your mongo service, as you can configure as to how much memory it will use for its heap - as its defaults may not be sufficient for your usecase.
